jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -34,31 +40,75 @@
  */
 public class ReproducibleLayerBuilder {
 
+  /**
+   * Holds a list of {@link TarArchiveEntry}s with unique extraction paths. The list also includes
+   * all parent directories for each extraction path.
+   */
+  private static class UniqueTarArchiveEntries {
```

Comment:
Why not just use a `LinkedHashMap` or something instead of making a custom data structure for it?

Reply:
The `LinkedHashMap` would require creating a custom wrapper that uses the `TarArchiveEntry#getName` as the hash object, which seemed more confusing than just having a custom data structure for storing a list of unique `TarArchiveEntry`s